src/server/game/Entities/Pet/Pet.h

```diff
@@ -59,9 +59,11 @@ class TC_GAME_API Pet : public Guardian
         bool CreateBaseAtCreature(Creature* creature);
         bool CreateBaseAtCreatureInfo(CreatureTemplate const* cinfo, Unit* owner);
         bool CreateBaseAtTamed(CreatureTemplate const* cinfo, Map* map, uint32 phaseMask);
-        bool LoadPetFromDB(Player* owner, uint32 petentry = 0, uint32 petnumber = 0, bool current = false);
+        static std::pair<PetStable::PetInfo const*, PetSaveMode> GetLoadPetInfo(PetStable const& stable, uint32 petEntry, uint32 petnumber, bool current);
+        bool LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool current);
         bool IsLoading() const override { return m_loading;}
         void SavePetToDB(PetSaveMode mode);
+        void FillPetInfo(PetStable::PetInfo* petInfo) const;
```

Comment:
```suggestion
        void FillPetInfo(PetStable::PetInfo& petInfo) const;
```

Reply:
Intended to make it clear at call site that this function modifies its parameter

Reply:
unsure how a non-const pointer is clearer than a non-const ref in that regard

Reply:
(i am very much not-fond of the core's tendency to pass pointers around with an implicit guarantee that they will never be null)

Reply:
Its visible at call site, `FillPetInfo(&var)`